backend/libs/room-operations.ts

```diff
@@ -137,17 +161,28 @@ export const joinExistingRoom = async (
     return updateWatcher(room, tableName, newWatcher, dynamoDb);
 };
 
-export const ensureRoomJoined = (
+/**
+ * Checks the user permission to modify the room the syncIntent
+ * Update it
+ *
+ * @param room The room to update
+ * @param tableName The name of the dynamodb table
+ * @param watcherConnectionString The connection id of the user trying to update the state
+ * @param syncIntent The new syncIntent for the room
+ */
+export const updateRoomSyncIntent = async (
     room: Room,
+    tableName: string,
     watcherConnectionString: string,
-) => {
-    if (!room.watchers[watcherConnectionString]) {
-        console.log(
-            '[WS-S] A media event was received for someone ho has not joined the room. Dropping',
-        );
-        throw new Error(
-            `The room was not joined by watcher ${watcherConnectionString}`,
-        );
+    syncIntent: SyncIntent,
+): Promise<Room> => {
+    if (
+        watcherConnectionString !== room.ownerId &&
+        syncIntent === SyncIntent.PLAY &&
+        syncIntentChanges(room, syncIntent)
+    ) {
+        throw new Error('Watcher is not authorized to perform this operation');
     }
-    return room;
+    room = assignRoomIntent(room, syncIntent);
```

Comment:
```suggestion
    room = assignRoomSyncIntent(room, syncIntent);
```
proposition pour rester cohérent